mergify_engine/check_api.py

```diff
@@ -148,20 +180,24 @@ def compare_dict(
 
 
 def check_need_update(
-    previous_check: github_types.GitHubCheckRun,
+    previous_check: github_types.GitHubCheckRunLight,
     expected_check: GitHubCheckRunParameters,
 ) -> bool:
     if compare_dict(
         typing.cast(typing.Dict[str, typing.Any], expected_check),
         typing.cast(typing.Dict[str, typing.Any], previous_check),
         ("head_sha", "status", "conclusion", "details_url"),
     ):
-        if previous_check["output"] == expected_check["output"]:
+        if previous_check["output"] is None and expected_check["output"] is None:
```

Comment:
a little bit worried by this being changed, I imagine it's related to this PR?

Reply:
I will fix that separately, the condition has always been false, when I  added the missing Check-runs attributes, mypy complains.